app/models/approval/request.rb

```diff
@@ -28,6 +28,12 @@ def self.define_user_association
       self.requested_at = Time.current
     end
 
+    after_commit(on: :update) do
+      if self.previous_changes["state"][0] != "executed" && self.previous_changes["state"][1] == "executed"
```

Comment:
- `previous_changes["state"]` may be nil
- If `previous_changes["state"][1]` is "executed", `previous_changes["state"][0]` should not be  "executed" because this is *change*

Reply:
Thanks!